mlflow/system_metrics/system_metrics_monitor.py

```diff
@@ -134,6 +143,7 @@ def finish(self):
         """Stop monitoring system metrics."""
         if self._process is None:
             return
+        self.mlflow_logger.flush()
```

Comment:
Wouldn't it be better to flush at the very end of the `finish` function, once all monitoring processes have been stopped? Because in the very unlikely case, some process might record new metrics between flushing the metrics and stopping the processes😅